crates/turborepo-globwatch/src/lib.rs

```diff
@@ -124,15 +139,29 @@ impl GlobWatcher {
     /// and config changes driven by the same stream. This allows us to ensure
     /// that anything watching for filesystem is also propagating config changes
     #[tracing::instrument(skip(self))]
-    pub fn into_stream(
+    pub async fn into_stream(
         self,
         token: stop_token::StopToken,
-    ) -> impl Stream<Item = Result<Event, TimedOutError>> + Send + Sync + 'static + Unpin {
+    ) -> Result<
+        impl Stream<Item = Result<Event, TimedOutError>> + Send + Sync + 'static + Unpin,
+        Error,
+    > {
+        let Self {
+            setup_handle,
+            flush_dir,
+            ..
+        } = self;
         let flush_id = Arc::new(AtomicU64::new(1));
-        let flush_dir = Arc::new(self.flush_dir);
+        let flush_dir = Arc::new(flush_dir);
         let flush = Arc::new(Mutex::new(HashMap::<u64, oneshot::Sender<()>>::new()));
 
-        Box::pin(
+        // Wait until we have watched the flush directory, before we start processing
+        // events
+        setup_handle
+            .await
+            .expect("globwatch setup thread panicked")?;
```

Comment:
Great, so we can queue up flush requests and guarantee that the dir is watched so they don't get lost. And it looks like the time-to-first-request for startup is not affected either. :)

Reply:
Something like this:

```rust
        let setup_stream = setup_handle.into_stream().map(|_| WatcherCommand::Start);
        (
            UnboundedReceiverStream::new(self.stream).map(Either::Left),
            setup_stream
                .chain(UnboundedReceiverStream::new(self.config))
                .map(Either::Right),
        )
```